Tools/peg_generator/pegen/build.py

```diff
@@ -61,6 +62,8 @@ def compile_c_extension(
     extra_link_args = get_extra_flags("LDFLAGS", "PY_LDFLAGS_NODIST")
     if keep_asserts:
         extra_compile_args.append("-UNDEBUG")
+    if disable_optimization:
+        extra_compile_args.append("-O0")
```

Comment:
@kumaraditya303 any idea why this didn't appear to cause the Windows CI to barf?  your https://github.com/python/cpython/pull/31017 is doing the same with a platform specific check to add a /Od flag which is presumably something I should guard against here to avoid -O0?

Reply:
Ya, I am experimenting if disabling link optimizations and optimizations makes any difference on CI, I am seeing a 10% improvement at least on on i5 processor 8 cores (hyperthreading).
> I should guard against here to avoid -O0?
Yes I think.

Reply:
FWIW if things are built as debug builds (as I suspect CI might be? I haven't looked at the configs) I doubt this flag will help much as those are usually lower optimization. It helps a ton on default optimized posix builds as a subset of our buildbots use.